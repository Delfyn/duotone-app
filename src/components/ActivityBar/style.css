@keyframes shift-rightwards {
  0% { transform: translateX(-100%); }
  40% { transform: translateX(0); }
  60% { transform: translateX(0); }
  100% { transform: translateX(100%); }
}

.activity-bar {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  width: 100%;
  height: .25rem;
  z-index: 1000;
  background-color: var(--primary-color);
  animation-name: shift-rightwards;
  animation-duration: 1s;
  animation-timing-function: ease-in-out;
  animation-delay: .8s;
  animation-iteration-count: infinite;
  animation-direction: initial;
  animation-fill-mode: initial;
  animation-play-state: initial;
}
