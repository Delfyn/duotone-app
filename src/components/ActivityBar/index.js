/* @flow */

import React from 'react';
import './style.css';

const ActivityBar = ({ active = false }: { active?: boolean }) =>
  active ? <div className="activity-bar" /> : null;

export default ActivityBar;
