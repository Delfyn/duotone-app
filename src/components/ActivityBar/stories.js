/* @flow */

import React from 'react';
import { storiesOf } from '@kadira/storybook';
import ActivityBar from './index';

storiesOf('ActivityBar', module)
  .add('active', () => (
    <ActivityBar active />
  ))
  .add('inactive', () => (
    <ActivityBar />
  ));
