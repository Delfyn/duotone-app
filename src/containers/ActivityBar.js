/* @flow */

import { connect } from 'react-redux';
import ActivityBar from '../components/ActivityBar';

const mapStateToProps = state => ({
  active: state.ui.active,
});

const ActivityBarContainer = connect(
  mapStateToProps,
)(ActivityBar);

export default ActivityBarContainer;
