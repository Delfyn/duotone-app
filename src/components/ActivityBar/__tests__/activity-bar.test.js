import React from 'react';
import renderer from 'react-test-renderer';
import ActivityBar from '../index';

test('ActivityBar renders correctly', () => {
  const component = renderer.create(<ActivityBar />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('ActivityBar renders active state correctly', () => {
  const component = renderer.create(<ActivityBar active />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
