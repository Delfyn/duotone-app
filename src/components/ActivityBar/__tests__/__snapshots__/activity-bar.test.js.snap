exports[`test ActivityBar renders active state correctly 1`] = `
<div
  className="activity-bar" />
`;

exports[`test ActivityBar renders correctly 1`] = `null`;
